### PR TITLE
UN-2571 [FEAT] Allow users to directly create API deployments from prompt studio

### PR DESCRIPTION
**Pull request by jagadeeswaran-zipstack**
## What

This PR implements the ability for users to create API deployments directly from the prompt studio interface, streamlining the workflow from tool development to API deployment.

## Why

Previously, users had to manually:
1. Export their prompt studio tool
2. Create a new workflow
3. Configure endpoints
4. Add tool instances
5. Create API deployment

This feature eliminates these manual steps by providing a guided process directly from the prompt studio.

## How

**Frontend Implementation:**
- Added `CreateApiDeploymentFromPromptStudio` component with a step-by-step modal interface
- Integrated the component into the prompt studio header dropdown menu
- Created a two-step process: deployment details configuration and tool settings
- Added proper validation, error handling, and success messaging
- Implemented navigation options after successful deployment creation

**Backend Integration:**
- Utilizes existing APIs for tool export, workflow creation, and API deployment
- Automatically configures endpoints with API connection type
- Handles tool instance creation and metadata configuration
- Supports tools with and without configurable settings

**Key Features:**
- Auto-generated unique API names with timestamps
- Tool settings configuration using existing RJSF form layout
- PostHog analytics integration for tracking deployment creation
- Proper error handling with backend validation feedback
- Success screen with navigation to either the API deployment or workflow configuration

## Notes on Testing

**Frontend Testing:**
- Test the complete flow from prompt studio header dropdown
- Verify form validation for deployment details (required fields, API name pattern)
- Test tool settings configuration for tools with and without schemas
- Verify success screen navigation options work correctly
- Test error handling for various failure scenarios

**Backend Integration Testing:**
- Verify tool export functionality works correctly
- Test workflow creation with unique naming
- Confirm endpoint configuration with API connection type
- Test tool instance creation and metadata updates
- Verify API deployment creation with proper parameters

**Edge Cases:**
- Tools without configurable settings (should show "no settings" message)
- Invalid API names (should show validation errors)
- Backend failures at different steps (should show appropriate error messages)
- Network connectivity issues during the multi-step process

## Screenshots

The feature includes:
1. **Header Integration**: New "Create API Deployment" option in prompt studio dropdown
2. **Step 1**: Deployment details configuration form
3. **Step 2**: Tool settings configuration (if available)
4. **Success Screen**: Completion confirmation with navigation options

## Related Issues/PRs

- Implements UN-2571: Allow users to directly create API deployments right from prompt studio
- Builds upon existing API deployment and workflow management infrastructure
- Enhances the prompt studio user experience by reducing manual workflow steps

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+      console.warn("No function name available for schema fetch");
+      setToolSchema(null);
+      return;
+    }
+
+    setIsSchemaLoading(true);
+    try {
+      const response = await axiosPrivate({
+        method: "GET",
+        url: `/api/v1/unstract/${sessionDetails?.orgId}/tool_settings_schema/?function_name=${functionName}`,
```
Comment (vishnuszipstack) [inline frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx]: @jagadeeswaran-zipstack use base url in a variable and use it in repeated places.

File: frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+          tool_id: toolFunctionName || toolDetails.tool_id,
+          workflow_id: workflowId,
+        },
+      });
+
+      // Step 5: Update tool instance with settings if provided
+      if (toolSettings && Object.keys(toolSettings).length > 0) {
+        await axiosPrivate({
+          method: "PATCH",
+          url: `/api/v1/unstract/${sessionDetails?.orgId}/tool_instance/${toolInstanceResponse.data.id}/`,
```
Comment (athul-rs) [inline frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx]: @jagadeeswaran-zipstack Can we please use the `useRequestUrl` hook to create the URL.

**Review by athul-rs — CHANGES_REQUESTED** [review]
LGTM.
Left a comment, please resolve before mergeing

**Review by Deepak-Kesavan — APPROVED** [review]

File: frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx (hunk 1 above)
Comment (jaags-dev) [reply frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx]: Made the required changes.

File: frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx (hunk 2 above)
Comment (jaags-dev) [reply frontend/src/components/deployments/create-api-deployment-from-prompt-studio/CreateApiDeploymentFromPromptStudio.jsx]: Made the required changes.

**Review by vishnuszipstack — APPROVED** [review]
